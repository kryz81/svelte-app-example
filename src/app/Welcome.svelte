<script>
  // props
  export let user = "";

  // state/computed
  $: welcomeMessage = `Hello ${user}! Welcome in eCampus.`;
</script>

<div class="row">
  <div class="col-md-12">{welcomeMessage}</div>
</div>
