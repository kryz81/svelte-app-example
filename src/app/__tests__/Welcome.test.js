import { render, cleanup } from "@testing-library/svelte";

import Welcome from "../Welcome.svelte";

beforeEach(cleanup);

describe("Welcome", () => {
  it("renders component", () => {
    const { container } = render(Welcome, { props: { user: "Test User" } });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders welcome message", () => {
    const { getByText } = render(Welcome, { props: { user: "Test User" } });

    expect(getByText("Hello Test User! Welcome in eCampus."));
  });
});
