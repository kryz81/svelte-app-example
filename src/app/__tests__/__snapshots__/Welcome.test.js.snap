// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders component 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-md-12"
    >
      Hello Test User! Welcome in eCampus.
    </div>
  </div>
</div>
`;
